### DISCUSSION TITLE
Created an Open Source Project Using Aspire - Seeking Help for Docker Deployment

### DISCUSSION THREAD

**Discussion by neozhu** [microsoft/aspire#6696] [Show and tell]
Hi everyone!

I've been working on an open-source project using Aspire with .NET 9 Minimal API / OpenAPI combined with Blazor Assembly (WPA). It's built with a lot of the latest technologies and is still in early development.

You can check out the project here: [GitHub Repository](https://github.com/neozhu/cleanaspire/).
![blazorclient](https://github.com/user-attachments/assets/0d777780-6cd1-4eb7-86a7-6b6b1b7d7e65)

Currently, I'm trying to solve a deployment challenge. Specifically, I'm looking for assistance in creating a Dockerfile that will allow me to build this project into a Docker image and publish it to Docker Hub. From there, I plan to create a docker-compose file to manage deployment using Docker Portainer.

However, I'm not entirely sure if there are other approaches within the Aspire project that could simplify or improve the deployment process. I'm also quite new to some of these tools and would greatly appreciate any help, advice, or collaboration from others here.

If you've got experience with Aspire or deploying similar projects, I'd love to hear your thoughts or suggestions!

Thank you so much in advance!

**Comment by davidfowl** [comment]
You have multiple projects, so are you asking for multiple docker files? .NET has container publishing built in so you don't need a docker file to produce a container image, see https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container.


> However, I'm not entirely sure if there are other approaches within the Aspire project that could simplify or improve the deployment process. I'm also quite new to some of these tools and would greatly appreciate any help, advice, or collaboration from others here.

https://github.com/prom3theu5/aspirational-manifests might be useful here.

**Comment by neozhu** [comment]
Thank you for your information, I'll give it a try
